File: Konfigurator-aspnetcore-Backend/Konfigurator-aspnetcore-Backend/Controllers/GameController.cs
using Konfigurator_aspnetcore_Backend.Models;
using Konfigurator_aspnetcore_Backend.Models.DTOs;
using Konfigurator_aspnetcore_Backend.Services.Game;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Konfigurator_aspnetcore_Backend.Controllers
{
    [Route("[controller]")]
    [ApiController]
    public class GameController : ControllerBase
    {
        private readonly IGameService _gameService;

        public GameController(IGameService gameService)
        {
            _gameService = gameService;
        }

        [HttpGet("{gameMapId:int}")]
        public async Task<ActionResult<ServiceResponseDto<GetGameMapResponseDto>>> GetGameMap(int gameMapId)
        {
            var response = await _gameService.GetGameMap(gameMapId);
            if(response.Success)
                return Ok(response.Data);
            return BadRequest(response.Message);
        }

        [HttpGet("getGameMaps")]
        public async Task<ActionResult<ServiceResponseDto<GetAllGameMapIdDto>>> GetAllGameMapIds()
        {
            var response = await _gameService.GetAllGameMapIds();
            if (response.Success)
                return Ok(response.Data);
            return BadRequest(response.Message);
        }

        [HttpGet("getGraphic")]
        public async Task<ActionResult<ServiceResponseDto<byte[]>>> GetGraphic(string graphic)
        {
            var response = await _gameService.GetGraphic(graphic);
            if(response.Success)
                return Ok(response.Data);
            return BadRequest(response.Message);
        }

        [HttpGet("getHighscores")]
        public async Task<ActionResult<ServiceResponseDto<GetHighscoresDto[]>>> GetHighscores(int gameMapId)
        {
            var response = await _gameService.GetHighscores(gameMapId);
            if (response.Success)
                return Ok(response.Data);
            return BadRequest(response.Message);
        }

        [HttpPost("addHighscore")]
        public async Task<ActionResult<ServiceResponse>> AddHighscore(int gameMapId, string username, int score)
        {
            var response = await _gameService.AddHighscore(gameMapId, username, score);
            if (response.Success)
                return Ok(response.Success);
            return BadRequest(response.Message);
        }
    }
}

File: Konfigurator-nestjs-Backend/konfigurator-nestjs-backend/src/game/game.controller.ts
import { Controller, Get, Post, Res, Param, Query } from '@nestjs/common';
import { GameService } from './game.service';
import { Response } from 'express';

@Controller('game')
export class GameController {
  constructor(private readonly gameService: GameService) {}

  @Get('getHighscores?')
  async GetHighscore(@Res() response: Response, @Query('gameMapId') gameMapId: number) {
    const resp = await this.gameService.GetHighscores(gameMapId);

    if(resp.success) {
      return response.send(resp.data);
    }

    return response.status(400).send(resp.message);
  }

  @Get('getGameMaps')
  async GetAllGameMapId(@Res() response: Response) {
    const resp = await this.gameService.GetAllGameMapsIds();

    if(resp.success) {
      return response.send(resp.data);
    }

    return response.status(400).send(resp.message);
  }

  @Get('getGraphic?')
  async GetGraphic(@Res() response: Response, @Query('graphic') graphic:string) {
    const resp = await this.gameService.GetGraphic(graphic);

    if(resp.success) {
      return response.send('\"' + resp.data + '\"');
    }

    return response.status(400).send(resp.message);
  }

  @Get('/:gameMapId')
  async GetGameMap(@Res() response: Response, @Param('gameMapId') gameMapId: number) {
    const resp = await this.gameService.GetGameMap(gameMapId)

    if(resp.success) {
      return response.send(resp.data);
    }

    return response.status(400).send(resp.message);
  }

  @Post('addHighscore?')
  async addHighscore(@Res() response: Response, @Query('gameMapId') gameMapId: number, @Query('username') username: string, @Query('score') score: number) {
    console.log(gameMapId, username, score)
    const resp = await this.gameService.addHighscore(gameMapId,username,score);

    if(resp.success) {
      return response.send(resp.data);
    }

    return response.status(400).send(resp.message);
  }
}
